wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
OK. I see!
But why do you need to change this code at all ? 
PageLock#getCycle() is not used anywhere in the PR.
The thread is very useful to identify problems with long running requests at the moment. The thread stack trace is telling me much more than a long (the request cycle start time).
I'd prefer to add the RequestCycle as yet another field to PageLock than removing the thread. But I don't really see the need to change PageAccessSynchronizer at all.